devel/onetbb/files/patch-onetbb-older-platforms.diff

```diff
@@ -0,0 +1,513 @@
+# https://github.com/oneapi-src/oneTBB/pull/840
+
+From 58f529cdb582848f8f5b8de49129c14dbff0cff3 Mon Sep 17 00:00:00 2001
+From: barracuda156 <vital.had@gmail.com>
+Date: Tue, 15 Nov 2022 18:06:45 +0800
+Subject: [PATCH] Restore support for 32-bit platforms, add fixes for PPC
+
+---
+ cmake/compilers/AppleClang.cmake            |   4 +-
+ cmake/compilers/GNU.cmake                   |   8 +-
+ include/oneapi/tbb/concurrent_queue.h       |   6 +-
+ include/oneapi/tbb/task_group.h             |  26 +++-
+ src/tbb/def/mac32-tbb.def                   | 159 ++++++++++++++++++++
+ src/tbb/tools_api/ittnotify_config.h        |   8 +-
+ src/tbb/tools_api/ittnotify_static.c        |   4 +-
+ src/tbbmalloc/def/mac32-tbbmalloc.def       |  45 ++++++
+ src/tbbmalloc/def/mac64-tbbmalloc.def       |   3 +-
+ src/tbbmalloc_proxy/proxy_overload_osx.h    |   8 +-
+ test/common/doctest.h                       |   5 +-
+ test/conformance/conformance_allocators.cpp |   5 +-
+ 12 files changed, 265 insertions(+), 16 deletions(-)
+ create mode 100644 src/tbb/def/mac32-tbb.def
+ create mode 100644 src/tbbmalloc/def/mac32-tbbmalloc.def
+
+diff --git a/cmake/compilers/AppleClang.cmake b/cmake/compilers/AppleClang.cmake
+index 5ebbdbd1..f27e1382 100644
+--- a/cmake/compilers/AppleClang.cmake
++++ b/cmake/compilers/AppleClang.cmake
+@@ -1,4 +1,4 @@
+-# Copyright (c) 2020-2022 Intel Corporation
++# Copyright (c) 2020-2023 Intel Corporation
+ #
+ # Licensed under the Apache License, Version 2.0 (the "License");
+ # you may not use this file except in compliance with the License.
+@@ -37,7 +37,7 @@ if (CMAKE_OSX_ARCHITECTURES)
+ else()
+     set(_tbb_target_architectures "${CMAKE_SYSTEM_PROCESSOR}")
+ endif()
+-if ("${_tbb_target_architectures}" MATCHES "(x86_64|amd64|AMD64)") # OSX systems are 64-bit only
++if ("${_tbb_target_architectures}" MATCHES "(i.86|x86|x86_64|amd64|AMD64)") # Clang is not supported on PowerPC
```

Comment:
This change doesn't seem necessary for PowerPC?

Reply:
Yes, but we can support i386 as well.

Reply:
Ah okay, I read the PR title (which only mentions PPC) but not the commit message. I guess change the PR title to match.